internal/pkg/manifest/workload.go

```diff
@@ -624,6 +627,63 @@ func (i ImageWithHealthcheck) HealthCheckOpts() *ecs.HealthCheck {
 	return i.HealthCheck.healthCheckOpts()
 }
 
+// PlatformArgsOrString is a custom type which supports unmarshaling yaml which
+// can either be of type string or type PlatformArgs.
+type PlatformArgsOrString struct {
+	PlatformString *string
+	PlatformArgs   PlatformArgs
+}
+
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the PlatformArgsOrString
+// struct, allowing it to perform more complex unmarshaling behavior.
+// This method implements the yaml.Unmarshaler (v2) interface.
+func (p *PlatformArgsOrString) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	if err := unmarshal(&p.PlatformArgs); err != nil {
+		switch err.(type) {
+		case *yaml.TypeError:
+			break
+		default:
+			return err
+		}
+	}
+
+	if !p.PlatformArgs.isEmpty() {
+		if !p.PlatformArgs.bothSpecifiedOrNil() {
+			return errors.New(`fields 'osfamily' and 'architecture' must either both be specified or both be empty. For more info, see your workload's manifest documentation at https://aws.github.io/copilot-cli/`)
+		}
+		if err := exec.ValidateOS(p.PlatformArgs.OSFamily); err != nil {
+			return fmt.Errorf("validate OS: %w", err)
+		}
+		if err := exec.ValidateArch(p.PlatformArgs.Arch); err != nil {
+			return fmt.Errorf("validate arch: %w", err)
+		}
+		// Unmarshaled successfully to p.PlatformArgs, unset p.PlatformString, and return.
+		p.PlatformString = nil
+		return nil
+	}
+	if err := unmarshal(&p.PlatformString); err != nil {
+		return errUnmarshalPlatformOpts
+	}
+	if err := exec.ValidatePlatform(p.PlatformString); err != nil {
+		return fmt.Errorf("validate platform: %w", err)
+	}
+	return nil
+}
+
+// PlatformArgs represents the specifics of a target OS.
+type PlatformArgs struct {
+	OSFamily *string `yaml:"osfamily,omitempty"`
+	Arch     *string `yaml:"architecture,omitempty"`
```

Comment:
Can we make this:
```suggestion
	OSFamily *string `yaml:"os,omitempty"`
	Arch     *string `yaml:"arch,omitempty"`
```

Reply:
I thought it made sense to match the terminology elsewhere, so folks using the console immediately know what is expected here.

Reply:
[discussed offline; ok to leave as is]